library/core/src/time.rs

```diff
@@ -697,6 +697,44 @@ impl Duration {
         }
     }
 
+    /// The checked version of [`from_secs_f64`].
+    ///
+    /// [`from_secs_f64`]: Duration::from_secs_f64
+    ///
+    /// This constructor will return an `Err` if `secs` is not finite, negative or overflows `Duration`.
+    ///
+    /// # Examples
+    /// ```
+    /// #![feature(duration_checked_float)]
+    ///
+    /// use std::time::Duration;
+    ///
+    /// let dur = Duration::try_from_secs_f64(2.7);
+    /// assert_eq!(dur, Ok(Duration::new(2, 700_000_000)));
+    ///
+    /// let negative = Duration::try_from_secs_f64(-5.0);
+    /// assert!(negative.is_err());
+    /// ```
+    #[unstable(feature = "duration_checked_float", issue = "83400")]
+    #[inline]
```

Comment:
Do I put a `#[rustc_const_unstable(...)]` attribute here and on the other function, or is it fine without?

Reply:
`rustc_const_unstable` is for the case where the function is stable but the const-ness isn't. You don't need it for unstable functions.